### PR TITLE
Filtering har entries by URL pattern

### PR DESCRIPTION
**Pull request by kiturutin**

### CODE REVIEW

**Review by epic-alex-rodionov — APPROVED** [review]

File: src/main/java/com/browserup/harreader/model/HarLog.java [hunk 1]
```diff
@@ -123,4 +129,41 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(version, creator, browser, pages, entries, comment);
     }
+
+    /**
+     * Search the entire log for the most recent entry whose request URL matches the given <code>url</code>.
```
Comment (ericbeland) [inline src/main/java/com/browserup/harreader/model/HarLog.java:134]: Could we add a few realistic regexp examples to our doc?  I've seen a lot of cases where messed up regexps make users think things "aren't working" so examples might be helpful. Classes of example regexps that might be handy to remind users of: 1-a standard one that matches http/https like you have in your test.  2-One that allows for any parameters, 3-One that wildcards a UUID in the middle of a regexp, 4-One that wildcards host/port.

File: src/main/java/com/browserup/harreader/model/HarLog.java [hunk 2]
```diff
@@ -123,4 +129,41 @@ public boolean equals(Object o) {
...
+     *
+     * @param url Regular expression match of URL to find.
+     *            URLs are formatted as: scheme://host:port/path?querystring.
+     *            Port is not included in the URL if it is the standard port for the scheme.
+     *            Fragments (example.com/#fragment) should not be included in the URL.
+     *            If more than one URL found, return the most recently requested URL.
+     * @return <code>HarEntry</code> for the most recently requested URL matching the given <code>url</code> pattern.
+     */
+    public Optional<HarEntry> findEntry(Pattern url) {
```
Comment (dpowell) [inline src/main/java/com/browserup/harreader/model/HarLog.java]: We should name this something to indicate it is returning the entry with the greatest starting time (vs the first, or an arbitrary one).

`findMostRecentEntry(Pattern)` perhaps?

@rkuzsma you've been naming these - do you have a preferred pattern for this?

**Review by dpowell — APPROVED** [review]
This looks good to me, let's confirm the method name for the "most recent" entry that matches with @rkuzsma prior to merging, otherwise good to go.

**Review by ericbeland — APPROVED** [review]
👍

**Review by dpowell — APPROVED** [review]

**Comment by rkuzsma** [comment]
The naming is great, the implementation is great, the tests are great. 👍 👍  Nice work Kirill!!

**Pull request merged**